test/runner.js

```diff
@@ -65,6 +67,44 @@ files = files.reduce( ( memo, filePath ) => {
 
 	// Append individual file argument
 	if ( /\.jsx?$/i.test( filePath ) ) {
+		if ( ! filePath.includes( '/test/' ) ) {
+			// did we just forget the test directory?
+			const pathGuess = path.join(
+				path.dirname( filePath ),
+				'test',
+				path.basename( filePath )
+			);
+
+			console.warn( pathGuess );
+			if ( fs.existsSync( pathGuess ) ) {
+				console.warn(
+					chalk.red.bold( 'WARNING:' ),
+					chalk.yellow(
+						'It appears that you\'re trying to use the test runner ' +
+						'on a file under test, not the test file itself.\n'
+					),
+					chalk.green( 'Did you mean to run against…\n\t' ),
```

Comment:
I noticed a bit of whitespace inconsistency between this and the "instead of" line, because the latter is prefixed with newline rather than by the previous argument, and separate arguments are joined by a space. I like the space prefix, and think maybe it would be easier to read a couple spaces apart instead of tab characters, like...

```js
chalk.red.bold( 'WARNING:' ),
chalk.yellow(
	'It appears that you\'re trying to use the test runner ' +
	'on a file under test, not the test file itself.'
),
chalk.green( '\n Did you mean to run against…\n  ' ),
chalk.blue( pathGuess ),
chalk.green( '\n …instead of…\n  ' ),
chalk.blue( filePath )
```